demo.sh

```diff
@@ -47,19 +46,29 @@ TKN() {
  tkn -n ${NAMESPACE} "$@"
 }
 
-demo.validate() {
+demo.validate_tools() {
   info "validating tools"
 
   tkn version >/dev/null 2>&1 || err 1 "no tkn binary found"
-  oc version >/dev/null 2>&1 || err 1 "no oc binary found"
+  oc version --client >/dev/null 2>&1 || err 1 "no oc binary found"
   return 0
 }
 
-declare -r TASKS_DIR=tmp/tasks
+demo.validate_pipelinerun() {
+  local failed=0
+  local results=$(oc get pipelinerun.tekton.dev -n ${NAMESPACE} --output=jsonpath='{range .items[*]}{.metadata.name}={.status.conditions[*].type}{.status.conditions[*].status}{" "}{end}')
```

Comment:
please always quote "" variables e.g. `-n "$NAMESPACE" `

Reply:
you can make use of arrays to iterate ...  might want to consider something like this 

```
loca results=( $(oc get pipelineruns --template='
{{range .items -}}
  {{ $pr := .metadata.name -}}
  {{ $c := index .status.conditions 0 -}}
  {{ $pr }}={{ $c.type }}{{ $c.status }}
{{ end }}
') )

for pr in ${result[@]}; do echo $pr ... ; done  

```

Reply:
Alternatively, I can just extract the jsonpath into separate variable, which one do you prefer?